gobblin-utility/src/main/java/gobblin/util/SchedulerUtils.java

```diff
@@ -68,6 +76,35 @@ public boolean accept(File dir, String name) {
     }
   };
 
+  private static final PathFilter PROPERTIES_PATH_FILTER =        new PathFilter() {
+    @Override
+    public boolean accept(Path path) {
+      String fileExtension = path.getName().substring(path.getName().lastIndexOf('.') + 1);
+      return fileExtension.equalsIgnoreCase(JOB_PROPS_FILE_EXTENSION);
+    }
+  };
+
+  private static final PathFilter NON_PROPERTIES_PATH_FILTER = new PathFilter() {
+    @Override
+    public boolean accept(Path path) {
+      String fileExtension = path.getName().substring(path.getName().lastIndexOf('.') + 1);
+      return !fileExtension.equalsIgnoreCase(JOB_PROPS_FILE_EXTENSION);
+    }
+  };
+
+  /**
+   * Load job configuration from job configuration files stored in general file system,
+   * located by Path
+   */
+  public static List<Properties> loadGenericJobConfigs(Properties properties)
+      throws ConfigurationException, IOException {
+    List<Properties> jobConfigs = Lists.newArrayList();
+    loadGenericJobConfigsRecursive(jobConfigs, properties, getJobConfigurationFileExtensions(properties),
+        (new Path(properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_DIR_KEY))).toUri() );
```

Comment:
The reason why I passed the URI is related to the implementation of `loadGenericJobConfigsRecursive`, in which there will be a line : `FileSystem filesystem = FileSystem.get(uri, conf); `  

Since FIleSystem constructor doesn't accepet Path, I introduce URI in.